src/GraphQL/Types/InputObjectGraphType.cs

```diff
@@ -1,6 +1,109 @@
-﻿namespace GraphQL.Types
+using GraphQL.Builders;
+using GraphQL.Resolvers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace GraphQL.Types
 {
     public class InputObjectGraphType : ComplexGraphType<object>
     {
     }
+
+
+
+    public class InputObjectGraphType<TSourceType> : InputObjectGraphType
```

Comment:
What we have done elsewhere is reverse this inheritance.

```csharp
public class InputObjectGraphType<TSourceType> : ComplexGraphType<TSourceType> {}

public class InputObjectGraphType : InputObjectGraphType<object> {}
```

Though that will probably cause problems with how we're doing checks on `InputObjectGraphType` and a number of people (including yourself) will want to use the generic `InputObjectGraphType<T>`.

There is an open issue related to this (#335) that I would like to fix for the 2.0 release.  We can use an interface to represent the input object and use that where we are now using the `InputObjectGraphType`.  Similar to what you were attempting to do with the `IComplexGraphType` but it would be input specific.